File: util/src/main/java/com/greenops/util/datamodel/pipelinestatus/FailedStep.java
package com.greenops.util.datamodel.pipelinestatus;

public class FailedStep {
    private String step;
    private boolean deploymentFailed;
    private String brokenTest;
    private String brokenTestLog;

    public FailedStep(String step, boolean deploymentFailed, String brokenTest, String brokenTestLog) {
        this.step = step;
        this.deploymentFailed = deploymentFailed;
        this.brokenTest = brokenTest;
        this.brokenTestLog = brokenTestLog;
    }

    public String getStep() {
        return step;
    }

    public boolean isDeploymentFailed() {
        return deploymentFailed;
    }

    public String getBrokenTest() {
        return brokenTest;
    }

    public String getBrokenTestLog() {
        return brokenTestLog;
    }
}

File: workflowtrigger/pipelinestatus/failed_step.go
package pipelinestatus

import "gitlab.com/c0b/go-ordered-json"

// TODO: should be replaced with PipelineSpec from CRD
type FailedStep struct {
	Step             string `json:"step"`
	DeploymentFailed bool   `json:"deploymentFailed"`
	BrokenTest       string `json:"brokenTest"`
	BrokenTestLog    string `json:"brokenTestLog"`
}

func MarshallFailedStep(failedStep FailedStep) *ordered.OrderedMap {
	mapObj := ordered.NewOrderedMap()
	mapObj.Set("step", failedStep.Step)
	mapObj.Set("deploymentFailed", failedStep.DeploymentFailed)
	mapObj.Set("brokenTest", failedStep.BrokenTest)
	mapObj.Set("brokenTestLog", failedStep.BrokenTestLog)
	return mapObj
}
